src/libraries/Microsoft.Extensions.Configuration.Binder/src/ConfigurationBinder.cs

```diff
@@ -298,6 +298,11 @@ private static void BindProperty(PropertyInfo property, object instance, IConfig
                 return;
             }
 
+            if (config is null)
+            {
+                return;
+            }
+
```

Comment:
do we understand why this didn't repro earlier? it could mean there is some other bug as well

Reply:
@krwq this is understood. Please have a look at the @ericstj comment https://github.com/dotnet/runtime/pull/92384#pullrequestreview-1636607530 which is pointing at the change that caused this regression.

Reply:
Yep, the null check was removed